Sources/NIO/BaseSocket.swift

```diff
@@ -301,6 +301,29 @@ class BaseSocket: BaseSocketProtocol {
         }
         return sock
     }
+    
+    /// Cleanup the unix domain socket.
+    ///
+    /// Deletes the associated file if it exists and has socket type.
+    ///
+    /// - parameters:
+    ///     - unixDomainSocketPath: The pathname of the UDS.
+    /// - throws: An `IOError` if the operation failed.
+    static func cleanupSocket(unixDomainSocketPath: String) throws {
+        let sb: UnsafeMutablePointer<stat> = UnsafeMutablePointer<stat>.allocate(capacity: 1)
+        if lstat(unixDomainSocketPath, sb) == 0 {
+            // Only unlink the existing file if it is a socket
+            if sb.pointee.st_mode & S_IFSOCK == S_IFSOCK {
+                let res = unlink(unixDomainSocketPath)
```

Comment:
We need to create a wrapped version of `unlink` as well.

Reply:
created a very trivial wrapped version, but didn't fully follow the logic of which are made available in what platforms, e.g. `fstat` exists in Android too, but only defined under linux and and macOS/iOS/etc. `unlink` should be available across all platforms